packages/runtime/src/components/_internal/ImplWrapper/hooks/useSlotChildren.tsx

```diff
@@ -17,8 +17,20 @@ export function getSlotElements(
       return <ImplWrapper key={child.id} {...props} component={child} />;
     });
 
-    slotElements[slot] = function getSlot(slotProps) {
-      return slotChildren.map(child => React.cloneElement(child, { slotProps }));
+    slotElements[slot] = function getSlot(slotProps, slotFallback) {
+      const slotContext = {
+        renderSet: new Set(childrenMap[c.id][slot].map(child => child.id)),
```

Comment:
Why do we need Set here? Every component's id is unique. It can not be duplicated.

Reply:
Because we need to remove the id when it's hidden and check the remaining size, a set is a little easier and more efficient to remove an element than an array.

I've also tried to use a number as the counter, but some React double render mechanisms make it very unstable.